src/rgw/rgw_data_sync.cc

```diff
@@ -2955,8 +2956,9 @@ class CheckBucketShardStatusIsIncremental : public RGWReadBucketPipeSyncStatusCo
  public:
   CheckBucketShardStatusIsIncremental(RGWDataSyncCtx* sc,
                                       const rgw_bucket_sync_pair_info& sync_pair,
+                                      uint64_t gen,
```

Comment:
`CheckBucketShardStatusIsIncremental` is for the backward-compatability check, and will always check `gen=0` so the caller shouldn't need to pass that in